scripts/s3/migrate_to_external_accounts.py

```diff
@@ -0,0 +1,332 @@
+import logging
+import os
+import sys
+import urlparse
+
+from modularodm import Q
+
+from framework.mongo import database
+from framework.transactions.context import TokuTransaction
+
+from website import settings
+from website.app import init_app
+from website.models import User, Node
+from website.oauth.models import ExternalAccount
+from website.addons.s3 import settings as s3_settings
+from website.addons.s3 import utils
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+PROVIDER = 's3'
+PROVIDER_NAME = 'Amazon S3'
+ENCRYPT_UPLOADS = s3_settings.ENCRYPT_UPLOADS_DEFAULT
+
+
+def verify_user_settings_documents(user_document):
+    try:
+        assert('_id' in user_document)
+        assert('deleted' in user_document)
+        assert('owner' in user_document)
+        assert('access_key' in user_document)
+        assert('secret_key' in user_document)
+        assert(user_document.get('owner', None))
+    except AssertionError:
+        return False
+    else:
+        return True
+
+def verify_node_settings_document(document):
+    try:    
+        assert('_id' in document)
+        assert('deleted' in document)
+        assert('bucket' in document)
+        assert('owner' in document)
+        assert('user_settings' in document)
+        assert(document.get('owner', None))
+    except AssertionError:
+        return False
+    try:
+        assert('encrypt_uploads' in document)
+    except AssertionError:
+        try:
+            database['addons3nodesettings'].find_and_modify(
+                {'_id': document['_id']},
+                {
+                    '$set': {
+                        'encrypt_uploads': ENCRYPT_UPLOADS,
+                    }
+                }
+            )
+        except Exception:
+            return False
+    return True
+
+def migrate_to_external_account(user_settings_document):
+    user_id = utils.get_user_id(access_key=user_settings_document['access_key'], secret_key=user_settings_document['secret_key'])
+    user = User.load(user_settings_document['owner'])
+    if not user_id:
+        return (None, None, None)
+    else:
+        display_name = 'S3User<[...]>{}'.format(user_id[-10:])
+
+    new = False
+    try:
+        external_account = ExternalAccount.find(Q('provider_id', 'eq', user_id))[0]
```

Comment:
This was done mostly because error handling is more complicated for `find_one`, and this query is guaranteed to return at most one result.